indico/modules/rb_new/client/js/modules/admin/AdminArea.jsx

```diff
@@ -0,0 +1,232 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import roomsSpriteURL from 'indico-url:rooms_new.sprite';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {Button, Container, Grid, Icon, Item, Menu, Placeholder} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+import SpriteImage from '../../components/SpriteImage';
+import {selectors as configSelectors} from '../../common/config';
+import * as adminSelectors from './selectors';
+import * as adminActions from './actions';
+
+import './AdminArea.module.scss';
+
+
+class AdminArea extends React.Component {
+    static propTypes = {
+        locations: PropTypes.array.isRequired,
+        roomsSpriteToken: PropTypes.string.isRequired,
+        isFetchingLocations: PropTypes.bool.isRequired,
+        actions: PropTypes.exact({
+            fetchLocations: PropTypes.func.isRequired,
+        }).isRequired,
+    };
+
+    state = {
+        activeItem: null,
+        locations: [],
+    };
+
+    static getDerivedStateFromProps(props, prevState) {
```

Comment:
why not use the router to determine what page to render and which menu item is active?

Reply:
Yeah, good idea.